include/Note.h

```diff
@@ -64,15 +65,16 @@ enum Octaves
 	Octave_6,
 	Octave_7,
 	Octave_8,
+	Octave_9,	// incomplete octave, MIDI only goes up to G9
 	NumOctaves
-} ;
-
+};
 
+const int FirstOctave = -1;
 const int WhiteKeysPerOctave = 7;
 const int BlackKeysPerOctave = 5;
 const int KeysPerOctave = WhiteKeysPerOctave + BlackKeysPerOctave;
-const int NumKeys = NumOctaves * KeysPerOctave;
-const int DefaultKey = DefaultOctave*KeysPerOctave + Key_A;
+const int NumKeys = qMin(NumOctaves * KeysPerOctave, 128);	// limited to MIDI range
```

Comment:
Hm, if 128 was a constant defined elsewhere, it could still make sense like this, as it is more informative. But I don't see LMMS going back to lower range or going up over MIDI range any time soon, so you are probably right.